### PR TITLE
Added DatabaseHandler::insert_or_update_profiles and test

### PR DESCRIPTION
**Pull request by ghost**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [ ] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: tests/lib/ocpp/v201/test_database_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    if (select_stmt->step() != SQLITE_ROW) {
+        // TODO ...
+        return;
+    }
+
+    auto count = select_stmt->column_int(0);
+    ASSERT_EQ(count, 2);
+}
+
+} // namespace ocpp::v201
```
Comment (christopher-davis-afs) [inline tests/lib/ocpp/v201/test_database_handler.cpp]: Everest folks like newlines at the end of files.

File: tests/lib/ocpp/v201/test_database_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+};
+
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithNoData_InsertProfile) {
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{1,1});
+    std::string sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+    auto select_stmt = db_connection->new_statement(sql);
+    
+    if (select_stmt->step() != SQLITE_ROW) {
+        // TODO ...
```
Comment (christopher-davis-afs) [inline tests/lib/ocpp/v201/test_database_handler.cpp]: For these TODOs, do you think we could add a little more context regarding what we should be putting here eventually?

**Review by christopher-davis-afs — CHANGES_REQUESTED** [review]
Just a few nitpicks, but otherwise LGTM.

File: tests/lib/ocpp/v201/test_database_handler.cpp (hunk 2 above)
Comment (ghost) [reply tests/lib/ocpp/v201/test_database_handler.cpp]: I added a check to make sure a row was returned. If it is not then no data was saved so we should fail.

**Review by christopher-davis-afs — APPROVED** [review]

File: include/ocpp/v201/database_handler.hpp [hunk 3]
```diff
@@ -118,6 +118,9 @@ class DatabaseHandler : public common::DatabaseHandlerCommon {
 
     /// \brief Remove all metervalue entries linked to transaction with id \p transaction_id
     bool transaction_metervalues_clear(const std::string& transaction_id);
+
+    /// \brief Inserts or updates the given \p profile to CHARGING_PROFILES table
```
Comment (gberardi-pillar) [inline include/ocpp/v201/database_handler.hpp:122]: Are we using a connector_id or an evse_id?

**Review by folkengine — APPROVED** [review]

File: include/ocpp/v201/database_handler.hpp (hunk 3 above)
Comment (ghost) [reply include/ocpp/v201/database_handler.hpp:122]: Great catch. Updated to evse_id

**Review by gberardi-pillar — APPROVED** [review]

**Review by christopher-davis-afs — CHANGES_REQUESTED** [review]
Just realized that this doesn't update the status doc but does tackle a functional requirement.

**Comment by ghost** [comment]
> Just realized that this doesn't update the status doc but does tackle a functional requirement.

@christopher-davis-afs  I was holding off marking that until we get the full boot call completed.

**Review by christopher-davis-afs — APPROVED** [review]

**Pull request merged**